libknet/transport_sctp.c

```diff
@@ -303,7 +303,9 @@ int sctp_transport_tx_sock_error(knet_handle_t knet_h, int sockfd, int recv_err,
 			/* Don't hold onto the lock while sleeping */
 			pthread_rwlock_unlock(&knet_h->global_rwlock);
 			usleep(knet_h->threads_timer_res / 16);
-			pthread_rwlock_rdlock(&knet_h->global_rwlock);
+			if (pthread_rwlock_rdlock(&knet_h->global_rwlock) != 0) {
+				log_warn(knet_h, KNET_SUB_TRANSP_SCTP, "Unable to get read lock, results might be unpredictable!");
```

Comment:
Is logging warning really best (or only) thing to do? not-getting lock is just asking for a trouble, so maybe it may make sense to return error code?

Reply:
yeah you are right, it might be possible to return an error. I'll look into it. thanks!